cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -282,7 +296,8 @@ private String toCanonicalContext(List<TagValue> tagValues) {
         sb.append(TAG_DELIM);
       }
       first = false;
-      sb.append(tv.getTagName()).append(TAG_DELIM).append(tv.getValue() == null ? ANY_TAG_VALUE : tv.getValue());
+      sb.append(encodeTag(tv.getTagName())).append(TAG_DELIM)
+        .append(tv.getValue() == null ? ANY_TAG_VALUE : encodeTag(tv.getValue()));
```

Comment:
I would not - it may be different for different requests, some may not accept it, etc.